x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/model/rule_assets/prebuilt_rule_asset.ts

```diff
@@ -83,3 +105,31 @@ export const PrebuiltRuleAsset = BaseCreateProps.omit(BASE_PROPS_REMOVED_FROM_PR
       version: RuleVersion,
     })
   );
+
+/**
+ * Creates a Map of the fields that are upgradable during the Upgrade workflow, by type.
+ * Creating this Map dynamically, based on BaseCreateProps and TypeSpecificFields, ensures that we don't need to:
+ *  - manually add rule types to this Map if they are created
+ *  - manually add or remove any fields if they are added or removed to a specific rule type
+ *  - manually add or remove any fields if we decide that they should not be part of the upgradable fields.
+ */
+function createUpgradableRuleFieldsByTypeMap() {
+  const baseFields = Object.keys(
+    BaseCreateProps.omit(BASE_PROPS_REMOVED_FROM_PREBUILT_RULE_ASSET).shape
+  );
+
+  const specificTypesToOmit: readonly string[] = TYPE_SPECIFIC_FIELDS_TO_OMIT;
+
+  return new Map(
+    TypeSpecificCreateProps.options.map((option) => {
+      const typeName = option.shape.type.value;
+      const typeSpecificFields = Object.keys(option.shape).filter(
+        // Filter out type-specific fields that should not be part of the upgradable fields
+        (field) => !specificTypesToOmit.includes(field)
+      );
+      return [typeName, [...baseFields, ...typeSpecificFields]];
```

Comment:
Another thought, maybe we should derive the upgradeable fields from the `PrebuiltRuleAsset` schema minus `rule_id`?